marklogic/src/main/scala/quasar/physical/marklogic/xml/package.scala

```
@@ -62,14 +62,16 @@ package object xml {
     *   }
     * }
     */
-  def toData(elem: Elem, config: KeywordConfig): Data = {
+  def toData(elem: Elem, config: KeywordConfig): Option[Data] = {
     def impl(nodes: Seq[Node], m: Option[MetaData]): Data = nodes match {
       case Seq() =>
         m.cata(attrsAndText(_,  ""), _str(""))
       case LeafText(txt) =>
         m.cata(attrsAndText(_, txt), _str(txt))
       case xs =>
-        val childrenByName = elements(xs) groupBy qualifiedName
+        val childrenAndLabel = (elements(xs) map withQualifiedName).toList.unite
+        val childrenByName = childrenAndLabel.groupBy(_._2).mapValues(elems => elems.map(_._1))
```

@jsantos17 I think we'll want to handle decoding this in `data.decodeXml`, this function here is intended to be purely for embedding `XML` in `Data` and should not be aware of any special encoding.

In `decodeXml` you have access to `MonadErrMsgs` where you can raise an error if an `ejson:key` element is missing the `key-id` attribute.